lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -1631,14 +1642,31 @@ std::size_t CBoostedTreeFactory::estimateMemoryUsageForTrain(std::size_t numberR
             : computeEta(numberColumns))};
     CScopeBoostedTreeParameterOverrides<std::size_t> overrides;
     overrides.apply(m_TreeImpl->m_Hyperparameters.maximumNumberTrees(), maximumNumberTrees);
-    std::size_t result{m_TreeImpl->estimateMemoryUsageTrain(numberRows, numberColumns)};
-    return result;
+    return m_TreeImpl->estimateMemoryUsageForTrain(numberRows, numberColumns);
 }
 
 std::size_t
 CBoostedTreeFactory::estimateMemoryUsageForTrainIncremental(std::size_t numberRows,
                                                             std::size_t numberColumns) const {
-    return m_TreeImpl->estimateMemoryUsageTrainIncremental(numberRows, numberColumns);
+    std::size_t maximumNumberTrees{this->mainLoopMaximumNumberTrees(
+        m_TreeImpl->m_Hyperparameters.eta().fixed()
+            ? m_TreeImpl->m_Hyperparameters.eta().value()
+            : computeEta(numberColumns))};
+    CScopeBoostedTreeParameterOverrides<std::size_t> overrides;
+    overrides.apply(m_TreeImpl->m_Hyperparameters.maximumNumberTrees(), maximumNumberTrees);
+    return m_TreeImpl->estimateMemoryUsageForTrainIncremental(numberRows, numberColumns);
```

Comment:
Should this code be moved to m_TreeImpl?

Reply:
The tricky part is that `mainLoopMaximumNumberTrees` logic relates to `CBoostedTreeFactory`. This sort of configuration seems to naturally belong on the factory, so I'm not sure I'd want to move that. `estimateMemoryUsageForTrainIncremental` does implement the bulk of memory estimation so I feel like this is probably ok.